### PR TITLE
Use new parent pom version

### PR DESCRIPTION
**Pull request by timja**
Partially fixes: https://github.com/jenkinsci/bom/pull/206
Other fix is in workflow-cps-global-lib

cc @basil @jglick

### CODE REVIEW

**Comment by basil** [comment]
>Partially fixes

Unclear to me what problem we are trying to solve or how this PR would solve it.

This plugin is currently using `plugin-pom` 3.57, which depends on `jenkins-test-harness` 2.61, which depends on `jenkins-test-harness-htmlunit` 2.36.0-1, which depends on `commons-text` 1.7, which depends on `commons-lang3` 3.9. `plugin-pom` 4.0-beta-6 depends on `jenkins-test-harness` 2.62, which depends on `jenkins-test-harness-htmlunit` 2.36.0-1, which depends on `commons-text` 1.7, which also depends on `commons-lang3` 3.9. So this change does not affect this plugin's dependency tree with regard to the `commons-lang3` dependency.

The error from [the failed run](https://ci.jenkins.io/job/Tools/job/bom/job/dependabot%2Fmaven%2Forg.jenkins-ci.plugins-timestamper-1.11.2/2/consoleFull) was:

```
Require upper bound dependencies error for org.apache.commons:commons-lang3:3.7 paths to dependency are:
+-io.jenkins.tools.bom:sample:7-rc308.6276ccd81655
  +-org.jenkins-ci.plugins.workflow:workflow-cps-global-lib:2.15
    +-org.apache.commons:commons-lang3:3.7
and
+-io.jenkins.tools.bom:sample:7-rc308.6276ccd81655
  +-org.jenkins-ci.plugins:timestamper:1.11.2
    +-org.apache.commons:commons-lang3:3.9
and
+-io.jenkins.tools.bom:sample:7-rc308.6276ccd81655
  +-org.jenkins-ci.main:jenkins-test-harness:2.57
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.31-2
      +-org.apache.commons:commons-lang3:3.7
and
+-io.jenkins.tools.bom:sample:7-rc308.6276ccd81655
  +-org.jenkins-ci.main:jenkins-test-harness:2.57
    +-org.jenkins-ci.main:jenkins-test-harness-htmlunit:2.31-2
      +-org.apache.commons:commons-text:1.3
        +-org.apache.commons:commons-lang3:3.7
```

Apart from the `workflow-cps-global-lib` dependency on `commons-lang3` 3.7 that has already been discussed, the other problematic dependency on `commons-lang3` 3.7 is `io.jenkins.tools.bom:sample`'s dependency on `plugin-pom` 3.55, which depends on `jenkins-test-harness` 2.57, which depends on `jenkins-test-harness-htmlunit` 2.31-2, which depends on `commons-lang3` 3.7 both directly and through `commons-text` 1.3. It seems to me that `io.jenkins.tools.bom:sample` should be upgraded from `plugin-pom` 3.55 to `plugin-pom` 3.56 (or later), which depends on `jenkins-test-harness` 2.60, which depends on `jenkins-test-harness-htmlunit` 2.36.0-1, which depends on `commons-text` 1.7, which depends on `commons-lang3` 3.9.

**Comment by timja** [comment]
right my mistake, it is being updated in https://github.com/jenkinsci/bom/pull/214 then should upgrade cleanly, although may need the workflow-cps-global-lib release as well

feel free to close or merge the PR.

**Comment by basil** [comment]
I think I'd prefer to wait to upgrade to `plugin-pom` 4.x until it is out of beta and has gained more adoption, unless there is a demonstrated need to upgrade sooner. Being an early adopter carries some degree of risk, and I don't have the time to take that on at present without a specific driver.

**Comment by timja** [comment]
> I think I'd prefer to wait to upgrade to `plugin-pom` 4.x until it is out of beta and has gained more adoption, unless there is a demonstrated need to upgrade sooner. Being an early adopter carries some degree of risk, and I don't have the time to take that on at present without a specific driver.

it's being released soon,

the main reason is so you don't need to manage jenkins core versions anymore, slf4j version is managed properly so when you bump your core version there's normally no enforcer issues to deal with

**Comment by jglick** [comment]
I would say updating to the 4.x parent is low risk—if your plugin still builds, you should be fine.

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.57</version>
+    <version>4.0-beta-6</version>
```
Comment (timja) [inline pom.xml:9]:
```suggestion
    <version>4.0</version>
```

**Comment by timja** [comment]
green now

**Pull request merged**